### PR TITLE
Add the internal debugging tool oeseal for SGX

### PR DESCRIPTION
**Pull request by mingweishih**
This PR adds `oeseal`, an internal debugging tool for SGX sealing API.
The tools supports seal/unseal operations given an input file and optionally writes the result to an output file.
In addition, the tool can optionally dumps the SGX key request data structure (including information such as cpusvn) associated with the sealed data.

Signed-off-by: Ming-Wei Shih <mishih@microsoft.com>

### CODE REVIEW

**Comment by mingweishih** [comment]
bors try

**Comment by mingweishih** [comment]
bors try

File: tests/tools/oeseal/host/host.c [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+} command_t;
+
+static void _print_helper()
+{
+    printf("Usage: oeseal <command> [<args>]\n"
+           "command: seal, unseal\n"
+           "args:\n"
+           "  -h, --help: Helper message\n"
+           "  -e, --enclave: Enclave binary\n"
+           "  -p, --policy: Sealing policy (product, unique)\n"
```
Comment (anakrish) [inline tests/tools/oeseal/host/host.c:40]: The produce id is hard-coded isn't it? In that case, does the "product" policy help with debugging?

**Review by radhikaj — APPROVED** [review]

File: tests/tools/oeseal/host/host.c (hunk 1 above)
Comment (mingweishih) [reply tests/tools/oeseal/host/host.c:40]: Correct. We can remove policy option.

Comment (mingweishih) [reply tests/tools/oeseal/host/host.c]: removed the policy option.

**Review by anakrish — APPROVED** [review]

**Pull request merged**